source/Screen.cpp

```diff
@@ -180,3 +180,11 @@ Point Screen::BottomRight()
 {
 	return Point(.5 * WIDTH, .5 * HEIGHT);
 }
+
+
+
+void Screen::SetDimensionsInternal(int width, int height)
```

Comment:
This seems like a code smell. 

Also, does the `Internal` mean anything in this context?

Reply:
I'm not proud of this code, but it was the most lightweight solution I could think of. 

Basically, every shader in the game uses the `Screen` class to get the rendered window size, so that it can get the aspect ratio of the rendered content correct. I need to be able to change that size to indicate that the new aspect ratio of the target renderbuffer. Otherwise, anything drawn to the target renderbuffer will appear distorted. 

My solution was to temporarily modify the `Screen::Dimensions()` temporarily while the render buffer was active. The alternative solution is to modify every shader to track whether or not the render buffer is active, and use the renderbuffer aspect ratio directly. 

The point of the `Internal` nomenclature, and the `private:` scope was to prevent future programmers from inadvertently using this function instead of `SetRaw()` which is probably what they really want. 

I've spent quite a bit of time on discord discussing other possible solutions, and I would love to hear any better ideas.